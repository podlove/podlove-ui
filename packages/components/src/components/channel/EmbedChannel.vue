<template>
  <channel type="button" channel="embed" :a11y="$props.a11y">
    <embed-icon />
  </channel>
</template>

<script setup lang="ts">
import Channel from './components/Channel.vue'
import EmbedIcon from '../icons/Embed.vue'

defineProps<{
  a11y?: string | null
}>()
</script>
