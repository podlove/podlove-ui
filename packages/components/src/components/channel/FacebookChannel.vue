<template>
  <channel type="link" channel="facebook" :href="facebookLink" :a11y="$props.a11y">
    <facebook-icon :filled="$props.filled" />
  </channel>
</template>

<script lang="ts" setup>
import { computed } from 'vue';
import { addQueryParameter } from '@podlove/utils/url';
import Channel from './components/Channel.vue';
import FacebookIcon from '../icons/Facebook.vue';

const LINK = 'https://www.facebook.com/sharer/sharer.php';

const props = defineProps<{ link: string | null; filled: boolean; a11y?: string | null }>();

const facebookLink = computed(() => addQueryParameter(LINK, { u: props.link }));
</script>
