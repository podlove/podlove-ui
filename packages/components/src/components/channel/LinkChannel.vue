<template>
  <channel type="link" channel="link" :href="$props.link" :a11y="$props.a11y" :target="null">
    <link-icon :filled="$props.filled" />
  </channel>
</template>

<script lang="ts" setup>
import Channel from './components/Channel.vue';
import LinkIcon from '../icons/Link.vue';

defineProps<{
  filled?: boolean;
  a11y?: string;
  link: string | null;
}>();
</script>
