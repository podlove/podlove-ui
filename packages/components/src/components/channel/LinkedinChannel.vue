<template>
  <channel type="link" channel="linkedin" :a11y="$props.a11y" :href="linkedinLink">
    <LinkedinIcon :filled="$props.filled" />
  </channel>
</template>

<script lang="ts" setup>
import { computed } from 'vue';
import { addQueryParameter } from '@podlove/utils/url';
import Channel from './components/Channel.vue';
import LinkedinIcon from '../icons/Linkedin.vue';

const LINK = 'https://www.linkedin.com/shareArticle';

const props = defineProps<{
  link: string | null;
  text: string;
  a11y?: string | null;
  filled?: boolean;
}>();

const linkedinLink = computed(() =>
  addQueryParameter(LINK, { url: props.link, mini: true, summary: props.text })
);
</script>
