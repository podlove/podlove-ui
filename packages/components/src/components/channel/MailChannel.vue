<template>
  <channel type="link" channel="mail" :a11y="$props.a11y" :href="mailLink">
    <mail-icon :filled="$props.filled" />
  </channel>
</template>

<script lang="ts" setup>
import { computed } from 'vue';
import { queryParameter } from '@podlove/utils/url';
import Channel from './components/Channel.vue';
import MailIcon from '../icons/Mail.vue';

const LINK = 'mailto:';

const props = defineProps<{
  text: string;
  subject: string;
  filled?: boolean;
  a11y?: string | null;
}>();

const mailLink = computed(
  () => `${LINK}?${queryParameter({ body: props.text, subject: props.subject })}`
);
</script>
