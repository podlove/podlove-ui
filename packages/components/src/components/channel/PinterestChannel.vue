<template>
  <channel type="link" channel="pinterest" :a11y="$props.a11y" :href="pinterestLink">
    <pinterest-icon :filled="$props.filled" />
  </channel>
</template>

<script lang="ts" setup>
import { computed } from 'vue';
import { addQueryParameter } from '@podlove/utils/url';
import Channel from './components/Channel.vue';
import PinterestIcon from '../icons/Pinterest.vue';

const LINK = 'https://pinterest.com/pin/create/bookmarklet/';

const props = defineProps<{
  link: string | null;
  text: string | null;
  poster: string | null;
  a11y?: string | null;
  filled?: boolean;
}>();

const pinterestLink = computed(() =>
  addQueryParameter(LINK, {
    url: props.link,
    media: props.poster,
    is_video: false,
    description: props.text
  })
);
</script>
