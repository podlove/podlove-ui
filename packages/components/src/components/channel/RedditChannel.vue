<template>
  <channel type="link" channel="reddit" :a11y="$props.a11y" :href="redditLink">
    <reddit-icon :filled="$props.filled" />
  </channel>
</template>

<script lang="ts" setup>
import { computed } from 'vue';
import { addQueryParameter } from '@podlove/utils/url';
import Channel from './components/Channel.vue';
import RedditIcon from '../icons/Reddit.vue';

const LINK = 'http://reddit.com/submit';

const props = defineProps<{
  text: string | null;
  link: string | null;
  filled?: boolean;
  a11y?: string;
}>();

const redditLink = computed(() => addQueryParameter(LINK, { url: props.link, title: props.text }));
</script>
