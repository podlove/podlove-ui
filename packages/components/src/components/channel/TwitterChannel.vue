<template>
  <channel type="link" channel="twitter" :a11y="$props.a11y" :href="twitterLink">
    <twitter-icon :filled="$props.filled" />
  </channel>
</template>

<script lang="ts" setup>
import { computed } from 'vue';
import { addQueryParameter } from '@podlove/utils/url';
import Channel from './components/Channel.vue';
import TwitterIcon from '../icons/Twitter.vue';

const LINK = 'https://twitter.com/intent/tweet';

const props = defineProps<{
  text: string | null;
  a11y?: string;
  filled?: boolean;
}>();

const twitterLink = computed(() => addQueryParameter(LINK, { text: props.text }));
</script>
