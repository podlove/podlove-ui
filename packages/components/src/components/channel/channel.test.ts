import { describe, test, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import EmbedChannel from './EmbedChannel.vue';
import FacebookChannel from './FacebookChannel.vue';
import LinkChannel from './LinkChannel.vue';
import LinkedinChannel from './LinkedinChannel.vue';
import MailChannel from './MailChannel.vue';
import PinterestChannel from './PinterestChannel.vue';
import RedditChannel from './RedditChannel.vue';
import TwitterChannel from './TwitterChannel.vue';
import WhatsAppChannel from './WhatsAppChannel.vue';
import XingChannel from './XingChannel.vue';

const props = {
  link: 'http://foo.bar',
  subject: 'test subject',
  text: 'test text',
  a11y: 'test a11y',
  poster: 'http://foo.bar/test.jpg',
  filled: false
};

const components = {
  embed: EmbedChannel,
  facebook: FacebookChannel,
  link: LinkChannel,
  linkedin: LinkedinChannel,
  mail: MailChannel,
  pinterest: PinterestChannel,
  reddit: RedditChannel,
  twitter: TwitterChannel,
  whatsApp: WhatsAppChannel,
  xing: XingChannel
};

describe('Channel', () => {
  Object.entries(components).forEach(([channel, component]) => {
    describe(channel, () => {
      test('should be a Vue instance', async () => {
        const wrapper = await mount(component, { props });
        expect(wrapper.vm).toBeTruthy();
      });

      test('should render the channel', async () => {
        const wrapper = await mount(component, { props });

        expect(wrapper.element).toMatchSnapshot();
      });
    });
  });
});
