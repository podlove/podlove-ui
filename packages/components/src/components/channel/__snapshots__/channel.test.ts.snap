// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Channel > embed > should render the channel 1`] = `
<button
  class="inline-flex items-center flex-col m-2 cursor-pointer"
  data-v-5755b7e2=""
  filled="false"
  link="http://foo.bar"
  poster="http://foo.bar/test.jpg"
  subject="test subject"
  text="test text"
>
  <span
    aria-hidden="true"
    class="podlove-component--channel flex justify-center items-center rounded border-2 border-transparent border-solid w-[60px] h-[60px]"
    data-v-5755b7e2=""
  >
    
    <svg
      aria-hidden="true"
      class="podlove-component-icon"
      data-v-258dccf6=""
      fill="none"
      height="25"
      viewBox="0 0 24 24"
      width="25"
      xmlns="http://www.w3.org/2000/svg"
    >
      
      <path
        d="M24 10.935v2.131l-8 3.947v-2.23l5.64-2.783-5.64-2.79v-2.223l8 3.948zm-16 3.848l-5.64-2.783 5.64-2.79v-2.223l-8 3.948v2.131l8 3.947v-2.23zm7.047-10.783h-2.078l-4.011 16h2.073l4.016-16z"
        fill="currentColor"
      />
      
    </svg>
    
  </span>
  <span
    class="sr-only"
    data-v-5755b7e2=""
  >
    test a11y
  </span>
</button>
`;

exports[`Channel > facebook > should render the channel 1`] = `
<a
  class="inline-flex items-center flex-col m-2 cursor-pointer"
  data-v-5755b7e2=""
  href="https://www.facebook.com/sharer/sharer.php?u=http%3A%2F%2Ffoo.bar"
  poster="http://foo.bar/test.jpg"
  subject="test subject"
>
  test text
</a>
`;

exports[`Channel > link > should render the channel 1`] = `
<a
  class="inline-flex items-center flex-col m-2 cursor-pointer"
  data-v-5755b7e2=""
  href="http://foo.bar"
  poster="http://foo.bar/test.jpg"
  subject="test subject"
>
  test text
</a>
`;

exports[`Channel > linkedin > should render the channel 1`] = `
<a
  class="inline-flex items-center flex-col m-2 cursor-pointer"
  data-v-5755b7e2=""
  href="https://www.linkedin.com/shareArticle?mini=true&summary=test%20text&url=http%3A%2F%2Ffoo.bar"
  poster="http://foo.bar/test.jpg"
  subject="test subject"
>
  <span
    aria-hidden="true"
    class="podlove-component--channel flex justify-center items-center rounded border-2 border-transparent border-solid w-[60px] h-[60px]"
    data-v-5755b7e2=""
  >
    
    <svg
      aria-hidden="true"
      class="podlove-component-icon"
      data-v-258dccf6=""
      fill="none"
      height="28"
      viewBox="0 0 28 28"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      
      <path
        clip-rule="evenodd"
        d="M3.5 6.16667C3.5 4.72386 4.72386 3.5 6.16667 3.5H21.8333C23.2761 3.5 24.5 4.72386 24.5 6.16667V21.8333C24.5 23.2761 23.2761 24.5 21.8333 24.5H6.16667C4.72386 24.5 3.5 23.2761 3.5 21.8333V6.16667ZM6.16667 4.5C5.27614 4.5 4.5 5.27614 4.5 6.16667V21.8333C4.5 22.7239 5.27614 23.5 6.16667 23.5H21.8333C22.7239 23.5 23.5 22.7239 23.5 21.8333V6.16667C23.5 5.27614 22.7239 4.5 21.8333 4.5H6.16667Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
      <path
        d="M8.16832 19.8333H10.6683V10.6666H8.16832V19.8333Z"
        fill="currentColor"
      />
      <path
        d="M7.95999 8.13998C7.95999 8.95165 8.61332 9.60998 9.41832 9.60998C10.2242 9.60998 10.8767 8.95165 10.8767 8.13998C10.8767 7.32832 10.2233 6.66998 9.41832 6.66998C8.61332 6.66998 7.95999 7.32832 7.95999 8.13998Z"
        fill="currentColor"
      />
      <path
        d="M18.1683 19.8333H20.6683V14.2008C20.6683 9.82332 15.9983 9.98248 14.835 12.1375V10.6666H12.335V19.8333H14.835V15.1633C14.835 12.5691 18.1683 12.3566 18.1683 15.1633V19.8333Z"
        fill="currentColor"
      />
      
    </svg>
    
  </span>
  <span
    class="sr-only"
    data-v-5755b7e2=""
  >
    test a11y
  </span>
</a>
`;

exports[`Channel > mail > should render the channel 1`] = `
<a
  class="inline-flex items-center flex-col m-2 cursor-pointer"
  data-v-5755b7e2=""
  href="mailto:?body=test%20text&subject=test%20subject"
  link="http://foo.bar"
  poster="http://foo.bar/test.jpg"
>
  <span
    aria-hidden="true"
    class="podlove-component--channel flex justify-center items-center rounded border-2 border-transparent border-solid w-[60px] h-[60px]"
    data-v-5755b7e2=""
  >
    
    <svg
      aria-hidden="true"
      class="podlove-component-icon"
      data-v-258dccf6=""
      fill="none"
      height="28"
      viewBox="0 0 28 28"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      
      <path
        d="M4.5 6.5H23.5M4.5 6.5V21.5M4.5 6.5L11 13.5M23.5 6.5V21.5M23.5 6.5L17 13.5M23.5 21.5H4.5M23.5 21.5L17 13.5M4.5 21.5L11 13.5M11 13.5L14 16.3333L17 13.5"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
      
    </svg>
    
  </span>
  <span
    class="sr-only"
    data-v-5755b7e2=""
  >
    test a11y
  </span>
</a>
`;

exports[`Channel > pinterest > should render the channel 1`] = `
<a
  class="inline-flex items-center flex-col m-2 cursor-pointer"
  data-v-5755b7e2=""
  href="https://pinterest.com/pin/create/bookmarklet/?description=test%20text&is_video=false&media=http%3A%2F%2Ffoo.bar%2Ftest.jpg&url=http%3A%2F%2Ffoo.bar"
  subject="test subject"
>
  <span
    aria-hidden="true"
    class="podlove-component--channel flex justify-center items-center rounded border-2 border-transparent border-solid w-[60px] h-[60px]"
    data-v-5755b7e2=""
  >
    
    <svg
      aria-hidden="true"
      class="podlove-component-icon"
      data-v-258dccf6=""
      fill="none"
      height="28"
      viewBox="0 0 28 28"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      
      <path
        clip-rule="evenodd"
        d="M14 25C20.0751 25 25 20.0751 25 14C25 7.92487 20.0751 3 14 3C7.92487 3 3 7.92487 3 14C3 20.0751 7.92487 25 14 25ZM4 13.9892C4 8.4725 8.48352 4 14.0146 4L13.9879 4.02167C19.5073 4.02167 24 8.49167 24 14.0108C24 19.5292 19.5073 24 13.9879 24C12.9487 24 11.9604 23.8417 11.0223 23.5542C11.4349 22.8925 12.049 21.8033 12.2737 20.9325L12.8994 18.5592C13.2369 19.1842 14.1892 19.7083 15.2034 19.7083C18.2325 19.7083 20.4103 16.9358 20.4103 13.49C20.4103 10.1933 17.7087 7.72083 14.2352 7.72083C9.91454 7.72083 7.6189 10.6175 7.6189 13.7642C7.6189 15.2375 8.40249 17.0608 9.65139 17.6358C9.8427 17.7242 9.94211 17.6858 9.98638 17.4983C10.0206 17.3617 10.1902 16.675 10.2654 16.3625C10.288 16.2625 10.2771 16.175 10.1944 16.075C9.78088 15.5767 9.45174 14.6517 9.45174 13.7908C9.45174 11.5933 11.1217 9.45833 13.9703 9.45833C16.4306 9.45833 18.154 11.1317 18.154 13.5158C18.154 16.2175 16.7839 18.0883 15.0062 18.0883C14.0246 18.0883 13.2853 17.2783 13.5234 16.2842C13.6254 15.8565 13.7614 15.4167 13.8937 14.9888C14.1288 14.2286 14.3521 13.5062 14.3521 12.9575C14.3521 12.1917 13.9394 11.5508 13.084 11.5508C12.0807 11.5508 11.2729 12.5908 11.2729 13.9767C11.2729 14.8608 11.5728 15.4608 11.5728 15.4608C11.5728 15.4608 10.5812 19.6442 10.3982 20.425C10.1977 21.2883 10.2763 22.5 10.364 23.2908C6.63815 21.8367 4 18.2217 4 13.9892Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
      
    </svg>
    
  </span>
  <span
    class="sr-only"
    data-v-5755b7e2=""
  >
    test a11y
  </span>
</a>
`;

exports[`Channel > reddit > should render the channel 1`] = `
<a
  class="inline-flex items-center flex-col m-2 cursor-pointer"
  data-v-5755b7e2=""
  href="http://reddit.com/submit?title=test%20text&url=http%3A%2F%2Ffoo.bar"
  poster="http://foo.bar/test.jpg"
  subject="test subject"
>
  <span
    aria-hidden="true"
    class="podlove-component--channel flex justify-center items-center rounded border-2 border-transparent border-solid w-[60px] h-[60px]"
    data-v-5755b7e2=""
  >
    
    <svg
      aria-hidden="true"
      class="podlove-component-icon"
      data-v-258dccf6=""
      fill="none"
      height="28"
      viewBox="0 0 28 28"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      
      <path
        d="M5.11208 16.5848C5.05458 16.8493 5.02487 17.1215 5.02487 17.3965C5.02487 20.6959 9.24328 23.3849 14.4296 23.3849C19.6139 23.3849 23.8323 20.6978 23.8323 17.3965C23.8323 17.1339 23.8046 16.8733 23.7538 16.6212L23.7394 16.5906C23.7193 16.5379 23.7116 16.4851 23.7116 16.4324C23.4222 15.3064 22.6412 14.2849 21.5114 13.4589C21.4682 13.4435 21.427 13.4215 21.3906 13.3918C21.3657 13.3726 21.3475 13.3515 21.3264 13.3304C19.6091 12.1479 17.1521 11.4062 14.4305 11.4062C11.7329 11.4062 9.29982 12.1335 7.58256 13.295C7.56914 13.3122 7.55477 13.3266 7.5356 13.3419C7.49823 13.3736 7.45511 13.3994 7.41103 13.4138C6.25532 14.2399 5.45419 15.27 5.1552 16.4066C5.1552 16.4621 5.14179 16.5158 5.11975 16.5704L5.11208 16.5848ZM14.4775 21.7989C12.7583 21.7989 11.548 21.4261 10.7756 20.6518C10.6098 20.4851 10.6098 20.2139 10.7756 20.0462C10.9443 19.888 11.2164 19.888 11.3841 20.0462C11.9878 20.6489 12.9988 20.9498 14.4775 20.9498C15.9552 20.9498 16.9604 20.6623 17.5622 20.0596C17.7203 19.9024 17.9935 19.9024 18.165 20.0596C18.3231 20.2321 18.3231 20.5052 18.165 20.6777C17.3897 21.452 16.1861 21.8257 14.4641 21.8257L14.4775 21.7989ZM11.0219 14.5417C10.146 14.5417 9.41482 15.2748 9.41482 16.1488C9.41482 17.0208 10.146 17.73 11.0219 17.73C11.8978 17.73 12.604 17.0208 12.604 16.1488C12.604 15.2748 11.8958 14.5417 11.0219 14.5417ZM17.9532 14.5417C17.0773 14.5417 16.3452 15.2748 16.3452 16.1488C16.3452 17.0208 17.0773 17.73 17.9532 17.73C18.8291 17.73 19.5354 17.0208 19.5354 16.1488C19.5354 15.2748 18.8272 14.5417 17.9532 14.5417ZM22.3115 13.0132C23.3053 13.8115 24.0365 14.7583 24.4092 15.7961C24.8405 15.4741 25.0982 14.9682 25.0982 14.4076C25.0982 13.4474 24.3172 12.6673 23.3561 12.6673C22.9737 12.6673 22.6105 12.7909 22.3115 13.0152V13.0132ZM5.59793 12.6692C4.63677 12.6692 3.85671 13.4512 3.85671 14.4114C3.85671 14.9318 4.08575 15.4157 4.47386 15.7425C4.85813 14.7085 5.59698 13.7751 6.59456 12.9864C6.30515 12.7852 5.95921 12.6711 5.59889 12.6711V12.6692H5.59793ZM14.4305 24.2444C8.77084 24.2444 4.16816 21.1731 4.16816 17.4003C4.16816 17.1368 4.1902 16.879 4.23332 16.6251C3.4734 16.1526 3 15.3199 3 14.4076C3 12.974 4.17583 11.8077 5.61135 11.8077C6.25724 11.8077 6.86001 12.0435 7.3334 12.4613C9.112 11.3199 11.5087 10.6012 14.1488 10.5524L15.8852 5.25874L16.2724 5.34978C16.2724 5.34978 16.2877 5.34978 16.2877 5.35169L20.3346 6.30328C20.6642 5.53856 21.4251 5 22.3134 5C23.4912 5 24.4514 5.96213 24.4514 7.14083C24.4514 8.32144 23.4902 9.28165 22.3134 9.28165C21.1367 9.28165 20.1765 8.31953 20.1765 7.14466L16.4861 6.2707L15.0802 10.561C17.6197 10.6616 19.9168 11.3794 21.6388 12.4977C22.1122 12.053 22.7351 11.8087 23.3954 11.8087C24.8309 11.8087 26 12.9711 26 14.4066C26 15.3524 25.4825 16.2139 24.6795 16.673C24.7073 16.9173 24.736 17.1464 24.736 17.3907C24.7216 21.1645 20.1305 24.2358 14.4631 24.2358L14.4305 24.2444ZM22.2684 5.85767C21.5593 5.85767 20.9843 6.43169 20.9843 7.13987C20.9843 7.84709 21.5593 8.42398 22.2684 8.42398C22.9699 8.42398 23.5429 7.84901 23.5429 7.14562C23.5429 6.44319 22.9699 5.86917 22.2521 5.86917L22.2684 5.85767Z"
        fill="currentColor"
      />
      
    </svg>
    
  </span>
  <span
    class="sr-only"
    data-v-5755b7e2=""
  >
    test a11y
  </span>
</a>
`;

exports[`Channel > twitter > should render the channel 1`] = `
<a
  class="inline-flex items-center flex-col m-2 cursor-pointer"
  data-v-5755b7e2=""
  href="https://twitter.com/intent/tweet?text=test%20text"
  link="http://foo.bar"
  poster="http://foo.bar/test.jpg"
  subject="test subject"
>
  <span
    aria-hidden="true"
    class="podlove-component--channel flex justify-center items-center rounded border-2 border-transparent border-solid w-[60px] h-[60px]"
    data-v-5755b7e2=""
  >
    
    <svg
      aria-hidden="true"
      class="podlove-component-icon"
      data-v-258dccf6=""
      fill="none"
      height="28"
      viewBox="0 0 28 28"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      
      <path
        d="M18.3016 5.83337C19.4766 5.83337 20.5342 6.3034 21.2392 7.12594C22.1793 6.89093 23.1193 6.65592 23.9419 6.06839C23.5893 7.00843 23.0018 7.83098 22.0618 8.301C23.0018 8.18349 23.7068 7.94848 24.4119 7.71347C23.8244 8.53601 23.1193 9.24105 22.4143 9.82858C22.4143 9.94608 22.4143 10.1811 22.4143 10.2986C22.4143 15.8214 18.1841 22.1667 10.5462 22.1667C8.19605 22.1667 5.96344 21.4617 4.08334 20.2866C4.43586 20.4041 4.78838 20.4041 5.1409 20.4041C7.1385 20.4041 8.90109 19.8166 10.3112 18.6415C8.54857 18.524 6.90349 17.349 6.43346 15.7039C6.66848 15.8214 6.90349 15.8214 7.1385 15.8214C7.49102 15.8214 7.84354 15.7039 8.19605 15.7039C6.43346 15.3514 5.02339 13.7063 5.02339 11.7087C5.61092 12.0612 6.19845 12.1787 6.90349 12.1787C5.84593 11.4737 5.02339 10.1811 5.02339 8.65352C5.02339 7.94848 5.2584 7.24345 5.61092 6.53841C7.72603 9.12354 10.7812 10.7686 14.1889 10.8861C14.1889 10.6511 14.1889 10.2986 14.1889 10.0636C14.1889 7.71347 16.069 5.83337 18.3016 5.83337Z"
        fill="transparent"
        stroke="currentColor"
        stroke-linejoin="round"
      />
      
    </svg>
    
  </span>
  <span
    class="sr-only"
    data-v-5755b7e2=""
  >
    test a11y
  </span>
</a>
`;

exports[`Channel > whatsApp > should render the channel 1`] = `
<a
  class="inline-flex items-center flex-col m-2 cursor-pointer"
  data-v-5755b7e2=""
  href="whatsapp://send?text=test%20text"
  link="http://foo.bar"
  poster="http://foo.bar/test.jpg"
  subject="test subject"
>
  <span
    aria-hidden="true"
    class="podlove-component--channel flex justify-center items-center rounded border-2 border-transparent border-solid w-[60px] h-[60px]"
    data-v-5755b7e2=""
  >
    
    <svg
      aria-hidden="true"
      class="podlove-component-icon"
      data-v-258dccf6=""
      fill="none"
      height="28"
      viewBox="0 0 28 28"
      width="28"
      xmlns="http://www.w3.org/2000/svg"
    >
      
      <path
        d="M3.26527 13.8373L3.26527 13.8352C3.24102 7.94305 8.04255 3.18341 13.9699 3.18341C19.8977 3.18341 24.7 7.94384 24.7 13.8119C24.7 19.6799 19.8977 24.4403 13.9699 24.4403C12.1855 24.4403 10.4888 23.9979 8.98486 23.2173C8.72894 23.0845 8.4249 23.0523 8.13965 23.1446L3.28614 24.716C3.22089 24.7372 3.15948 24.6748 3.18158 24.6099L4.75156 19.9981C4.85552 19.6927 4.81541 19.3649 4.66172 19.0965C3.77301 17.5444 3.26527 15.7351 3.26527 13.8373Z"
        stroke="currentColor"
      />
      <path
        d="M19.57 16.1461C19.4976 16.0496 19.3289 15.9773 19.0636 15.8326C18.7984 15.7121 17.5205 15.0852 17.2794 14.9887C17.0383 14.8923 16.8695 14.8682 16.7008 15.1093C16.532 15.3745 16.0256 15.9532 15.881 16.1219C15.7363 16.2907 15.5675 16.3148 15.3264 16.1943C15.0612 16.0737 14.2173 15.7844 13.2288 14.9164C12.4572 14.2413 11.9268 13.3733 11.7821 13.1322C11.6374 12.8669 11.758 12.7223 11.9026 12.6017C12.0232 12.4812 12.1679 12.2883 12.2884 12.1436C12.409 11.9989 12.4572 11.8784 12.5536 11.7096C12.6501 11.5408 12.6019 11.3962 12.5295 11.2515C12.4572 11.1309 11.9509 9.85306 11.7339 9.32261C11.5169 8.81628 11.2999 8.88861 11.1552 8.88861C11.0105 8.88861 10.8176 8.8645 10.6489 8.8645C10.4801 8.8645 10.1908 8.93684 9.94964 9.17795C9.70853 9.44317 9.03342 10.0701 9.03342 11.3479C9.03342 12.6258 9.97375 13.8555 10.0943 14.0243C10.2149 14.1931 11.9026 16.8935 14.5549 17.9303C17.2071 18.9671 17.2071 18.6295 17.6893 18.5813C18.1715 18.5331 19.2324 17.9544 19.4494 17.3516C19.6423 16.7729 19.6423 16.2666 19.57 16.1461Z"
        fill="currentColor"
      />
      
    </svg>
    
  </span>
  <span
    class="sr-only"
    data-v-5755b7e2=""
  >
    test a11y
  </span>
</a>
`;

exports[`Channel > xing > should render the channel 1`] = `
<a
  class="inline-flex items-center flex-col m-2 cursor-pointer"
  data-v-5755b7e2=""
  href="https://www.xing.com/spi/shares/new?url=http%3A%2F%2Ffoo.bar"
  poster="http://foo.bar/test.jpg"
  subject="test subject"
>
  test text
</a>
`;
